pkg/dns/aws/dns.go

```diff
@@ -242,83 +242,101 @@ func (m *Provider) change(record *iov1.DNSRecord, zone configv1.DNSZone, action
 		return fmt.Errorf("unsupported record type %s", record.Spec.RecordType)
 	}
 	// TODO: handle >0 targets
-	domain, target := record.Spec.DNSName, record.Spec.Targets[0]
+	domain, targets := record.Spec.DNSName, record.Spec.Targets
+	m.lock.Lock()
+	updatedRecords := sets.NewString(m.updatedRecords.List()...)
+	m.lock.Unlock()
 	if len(domain) == 0 {
 		return fmt.Errorf("domain is required")
 	}
-	if len(target) == 0 {
+	if len(targets) == 0 {
 		return fmt.Errorf("target is required")
 	}
 
+	var changes []*route53.Change
+
 	zoneID, err := m.getZoneID(zone)
 	if err != nil {
 		return fmt.Errorf("failed to find hosted zone for record: %v", err)
 	}
 
-	// Find the target hosted zone of the load balancer attached to the service.
-	targetHostedZoneID, err := m.getLBHostedZone(target)
-	if err != nil {
-		return fmt.Errorf("failed to get hosted zone for load balancer target %q: %v", target, err)
-	}
+	for _, target := range targets {
+		var key string
```

Comment:
Delete `var key string` and use `key :=` below.

Reply:
done